activemodel/lib/active_model/type/helpers/time_value.rb

```diff
@@ -22,7 +22,7 @@ def serialize(value)
         end
 
         def apply_seconds_precision(value)
-          return value unless precision && value.respond_to?(:usec)
+          return value unless precision && precision < 6 && value.respond_to?(:usec) && value.usec.positive?
```

Comment:
This condition is now longer (complex) enough.
We generally use early returning only if simple condition.

```ruby
if precision && precision < 6 && value.respond_to?(:usec) && value.usec.positive?
  ...
else
  value
end
```

Reply:
Fixed. Thanks!